lib/steps.js

```diff
@@ -706,6 +706,27 @@ When('I tap to continue', { timeout: -1 }, async () => {
   );
 });
 
+When(/I tap the "([^"]+)" tab/i, { timeout: -1 }, async ( tabId ) => {
+  return wrapPromiseWithTimeout(
+    scope.tapTab( scope, tabId ),
+    scope.timeout
+  )
+});
+
+When(/I tap the "([^"]+)" element$/i, { timeout: -1 }, async ( elemSelector ) => {
+  return wrapPromiseWithTimeout(
+    scope.tapElementById( scope, elemSelector, 2000 ),
```

Comment:
That's a separately defined step (that's just a different interface to this same code) that lets someone specify a timeout. IMO I didn't want to specify the timeout for every single step that I used this tap feature in, and unfortunately we do still need the time out. It's for browser events, so I can't imagine it being too flaky, since it's not based on server events. Ideally, the author / running shouldn't have to know that this timeout exists, so I don't want to make it super configurable elsewhere.

I looked into possibly using the [transition to end](https://stackoverflow.com/questions/51547994/wait-for-transition-to-end-in-puppeteer) value, but I couldn't actually get it to work for the one example I had (tapping on tabs in the tabbed interface), and I think what the actual transition that we should wait for will vary and I can't actually find?

Reply:
I'm willing to go with that choice.

Regarding catching transition events, you can go down the rabbit hole if you want, but it depends on the implementation being used and, yes, it would be specific to the situation. Also, tapping some things may not cause transitions, etc. It's pretty complicated even before you hit the difficulties in the code itself.

Reply:
I ran into some problems when testing this to make sure it worked, and ended up trying the transition events and it seems to work pretty well? I want to see if it'll work for the accessibility testing before merging (I remembered that the main reason I needed the wait was because I would tap the button to switch the tabs and then check for accessibility, but the page wouldn't have fully transitioned yet, so the accessibility check would check the colors with the in-progress fade-in. Those colors didn't have enough contrast, but also weren't the true colors of the page and so would be a flaky failure), but it's pretty smooth right now.

Reply:
Accessibility seems to work fine with the transition end stuff as well. I think it should be good.

Reply:
I'll have to see the code to see if it's general enough. To use it here, this would have to be able to detect the transition of tapping any element, not just a tab. If you want to just detect tab tapping transitions, that should probably be a separate Step just for tabs.

Reply:
Somehow my page wasn't updating with new code. I'll check it out now and see how it all flows.